identity/src/main/java/com/stripe/android/identity/navigation/IdentityNavGraph.kt

```diff
@@ -450,13 +463,52 @@ private fun DocumentUploadScreenContent(
     )
 }
 
+@ExperimentalMaterialApi
+/**
+ * Built a composable screen with ModalBottomSheetLayout
+ */
 private fun NavGraphBuilder.screen(
     route: IdentityTopLevelDestination.DestinationRoute,
     content: @Composable (NavBackStackEntry) -> Unit
 ) {
     composable(
         route = route.route,
-        arguments = route.arguments,
-        content = content
-    )
+        arguments = route.arguments
+    ) { navBackStackEntry ->
+        val bottomSheetViewModel = viewModel<BottomSheetViewModel>()
+        val bottomSheetState by bottomSheetViewModel.bottomSheetState.collectAsState()
```

Comment:
Potential alternative: 

For this kind of view-only states (show / hide a bottomsheet), if there's no viewmodel logic involved, to avoid lag it's fine to not do the composable -> viewmodel -> composable trip and keep the state view only. The receiver view can receive a callback to show / hide.

Reply:
Yeah I was planning to pass the callback to subviews and change the state too, but since the bottomsheet content will be sent through a custom url inside a html text that might appear any subview within a screen, I think it's more flexbile to not pass the callbacks to receivers, but rely on the `viewModel<>()` call to return us the viewModel whenever needed and call `viewModel.showBottomSheet()`

I think if we later migrated to a more portable [ModalBottomSheet](https://developer.android.com/jetpack/compose/components/bottom-sheets) the solution could be a bit more elegant